src/main/java/cr0s/warpdrive/block/TileEntityAirGenerator.java

```diff
@@ -35,10 +43,10 @@ public void updateEntity() {
 			}
 			return;
 		}
-		
+				
 		cooldownTicks++;
 		if (cooldownTicks > WarpDriveConfig.AIRGEN_AIR_GENERATION_TICKS) {
-			if (energy_consume(WarpDriveConfig.AIRGEN_ENERGY_PER_NEWAIRBLOCK, true)) {
+			if (energy_consume(WarpDriveConfig.AIRGEN_ENERGY_PER_NEWAIRBLOCK, true) && generatorEnabled) {
```

Comment:
You want to check the enabled state first as it's less CPU intensive than check energy consumption.